daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
am fine with that.
unfortunately, there can be only 1 variadic argument which we have to reserve for the network option (that libnetwork parses). All other driver options (including IPAM, network-driver and other drivers that will come in the future such as Service Discovery driver, etc...) should have explicit NetworkOption which takes `map[string]string`.

Should we have that done in this PR or should unblock this one in its current form (since we already have driver and network options mixed up currently) ? We could open an issue and address it as a proper API change PR. 

WDYT ?